### PR TITLE
Add launch commands

### PR DESCRIPTION
**Pull request by lantiga**
## What does this PR do?

Adds launch commands to the main package.

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda @carmocca

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM ! Finally :)

**Comment by lantiga** [comment]
In draft until we verify e2e

File: src/lightning/app/launcher/launcher.py [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+from functools import partial
+from multiprocessing import Process
+from typing import Callable, Dict, List, Optional, Tuple, TypedDict
+
+from lightning_utilities.core.imports import module_available
+
+ENABLE_MULTIPLE_WORKS_IN_DEFAULT_CONTAINER = bool(int(os.getenv("ENABLE_MULTIPLE_WORKS_IN_DEFAULT_CONTAINER", "0")))
+
+
+if module_available("lightning_app"):
```
Comment (Borda) [inline src/lightning/app/launcher/launcher.py:17]: this is not needed, the switch is on package level when we build it, you can see other commands...

File: src/lightning/app/launcher/launcher.py [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+    except (ImportError, ModuleNotFoundError):
+        ABLE_TO_RUN_APP_COMMANDS = False
+
+
+if ENABLE_MULTIPLE_WORKS_IN_DEFAULT_CONTAINER:
+    from lightning.app.launcher.lightning_hybrid_backend import CloudHybridBackend as CloudBackend
+else:
+    from lightning.app.launcher.lightning_backend import CloudBackend
+
+if True:
```
Comment (Borda) [inline src/lightning/app/launcher/launcher.py:65]: is this just for debugging?

File: src/lightning/app/launcher/launcher.py [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
+    app = load_app_from_file(entrypoint_file)
+
+    if module_available("lightning_app"):
+        from lightning_app.api.http_methods import _add_tags_to_api, _validate_api
+        from lightning_app.utilities.app_helpers import is_overridden
+        from lightning_app.utilities.commands.base import _commands_to_api, _prepare_commands
+    else:
+        from lightning.app.api.http_methods import _add_tags_to_api, _validate_api
+        from lightning.app.utilities.app_helpers import is_overridden
+        from lightning.app.utilities.commands.base import _commands_to_api, _prepare_commands
```
Comment (Borda) [inline src/lightning/app/launcher/launcher.py:116]:
```suggestion
    from lightning.app.api.http_methods import _add_tags_to_api, _validate_api
    from lightning.app.utilities.app_helpers import is_overridden
    from lightning.app.utilities.commands.base import _commands_to_api, _prepare_commands
```

File: src/lightning/app/launcher/launcher.py [hunk 4]
```diff
@@ -0,0 +1,480 @@
...
+    from lightning_app.utilities.load_app import extract_metadata_from_app, load_app_from_file
+    from lightning_app.utilities.proxies import WorkRunner
+    from lightning_app.utilities.redis import check_if_redis_running
+
+    try:
+        from lightning_app.utilities.app_commands import run_app_commands
+
+        ABLE_TO_RUN_APP_COMMANDS = True
+    except (ImportError, ModuleNotFoundError):
+        ABLE_TO_RUN_APP_COMMANDS = False
```
Comment (Borda) [inline src/lightning/app/launcher/launcher.py:36]:
```suggestion
```
this is not needed, the switch is on package level when we build it, you can see other commands...

File: src/lightning/app/launcher/launcher.py [hunk 5]
```diff
@@ -0,0 +1,480 @@
...
+    from lightning.app.utilities.load_app import extract_metadata_from_app, load_app_from_file
+    from lightning.app.utilities.proxies import WorkRunner
+    from lightning.app.utilities.redis import check_if_redis_running
+
+    try:
+        from lightning.app.utilities.app_commands import run_app_commands
+
+        ABLE_TO_RUN_APP_COMMANDS = True
+    except (ImportError, ModuleNotFoundError):
+        ABLE_TO_RUN_APP_COMMANDS = False
```
Comment (Borda) [inline src/lightning/app/launcher/launcher.py:57]:
```suggestion
from lightning.app import LightningFlow
from lightning.app.core import constants
from lightning.app.core.api import start_server
from lightning.app.core.queues import MultiProcessQueue, QueuingSystem
from lightning.app.storage.orchestrator import StorageOrchestrator
from lightning.app.utilities.cloud import _sigterm_flow_handler
from lightning.app.utilities.component import _set_flow_context, _set_frontend_context
from lightning.app.utilities.enum import AppStage
from lightning.app.utilities.exceptions import ExitAppException
from lightning.app.utilities.load_app import extract_metadata_from_app, load_app_from_file
from lightning.app.utilities.proxies import WorkRunner
from lightning.app.utilities.redis import check_if_redis_running

try:
    from lightning.app.utilities.app_commands import run_app_commands

    ABLE_TO_RUN_APP_COMMANDS = True
except (ImportError, ModuleNotFoundError):
    ABLE_TO_RUN_APP_COMMANDS = False
```

File: src/lightning/app/launcher/lightning_backend.py [hunk 6]
```diff
@@ -0,0 +1,542 @@
...
+
+if module_available("lightning_app"):
+    from lightning_app.storage import Mount
+else:
+    from lightning.app.storage import Mount
+
+if module_available("lightning_app"):
+    from lightning_app.utilities.exceptions import LightningPlatformException
+else:
+    from lightning.app.utilities.exceptions import LightningPlatformException
```
Comment (Borda) [inline src/lightning/app/launcher/lightning_backend.py:61]:
```suggestion
from lightning.app import LightningApp, LightningWork
from lightning.app.core.queues import QueuingSystem
from lightning.app.runners.backends.backend import Backend
from lightning.app.storage import Drive, Mount
from lightning.app.utilities.enum import make_status, WorkStageStatus, WorkStopReasons
from lightning.app.utilities.network import _check_service_url_is_ready, LightningClient
from lightning.app.utilities.exceptions import LightningPlatformException
```

File: src/lightning/app/launcher/lightning_hybrid_backend.py [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+    from lightning_app.runners.backends.mp_process import MultiProcessingBackend
+    from lightning_app.utilities.network import LightningClient
+
+else:
+    from lightning.app.core import constants
+    from lightning.app.core.queues import QueuingSystem
+    from lightning.app.launcher.lightning_backend import CloudBackend
+    from lightning.app.runners.backends.backend import Backend
+    from lightning.app.runners.backends.mp_process import MultiProcessingBackend
+    from lightning.app.utilities.network import LightningClient
```
Comment (Borda) [inline src/lightning/app/launcher/lightning_hybrid_backend.py:21]:
```suggestion
from lightning.app.core import constants
from lightning.app.core.queues import QueuingSystem
from lightning.app.launcher.lightning_backend import CloudBackend
from lightning.app.runners.backends.backend import Backend
from lightning.app.runners.backends.mp_process import MultiProcessingBackend
from lightning.app.utilities.network import LightningClient
```

File: tests/tests_app/cli/test_cmd_launch.py [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+from click.testing import CliRunner
+
+from lightning.app.core.queues import QueuingSystem
+from lightning.app.frontend.web import StaticWebFrontend
+from lightning.app.runners.runtime import load_app_from_file
+from lightning.app.testing.helpers import _RunIf, EmptyWork
+from lightning.app.utilities.network import find_free_network_port
+
+try:
+    from lightning.app.utilities.app_commands import run_app_commands
```
Comment (Borda) [inline tests/tests_app/cli/test_cmd_launch.py:18]: how does this happen a function is missing?

Comment (lantiga) [reply tests/tests_app/cli/test_cmd_launch.py:18]: Removing

File: src/lightning/app/launcher/launcher.py (hunk 2 above)
Comment (lantiga) [reply src/lightning/app/launcher/launcher.py]: This is to make qa happy with non-top imports

**Review by awaelchli — APPROVED** [review]

**Pull request merged**